genderize.py

```diff
@@ -0,0 +1,184 @@
+        for line in self.f:
+            if self.count >= 362:
+                text = line[0:86]
+                mf = text[:2].strip() # M,1M,?M, F,1F,?F, ?, =
+                #  =  <short_name> <long_name> 
+                name = text[2:29].lower().strip()
+                sortingFlag = text[29] # +,-; ignore +
+                unsortedf = text[30:85]
+                frequencies = split(unsortedf)
+            
+                if sortingFlag != '+':
+                            if mf == '=':
+                                continue
+                            else:
+                                if name.find('+') != -1:
+                                    names = [name.replace('+','-'), name.replace('+',' '), name.replace('+','').capitalize()]
+                                else:
+                                    names = [name]
+                                for name in names:
+                                    if name not in self.genderDict:
+                                        self.genderDict[name] = {}
+                                        self.genderDict[name][mf] = {}                                   
+                                    else:  
+                                        if mf not in self.genderDict[name]:
+                                            self.genderDict[name][mf] = {} 
+                                            
+                                    for country,freq in zip(self.countries,frequencies):
+                                        if freq != ' ':
+                                            norm_freq = int((float(self.pop[country]) * 1000) * 0.02 * (2 ** (int(freq,16) - 10)))
+                                            #norm_freq = int(((float(pop[country]))*1000) * (math.log(int(freq,16))))
+                                            # print(int(float(pop[country]) * 1000)) 
+                                            # print((math.log(int(freq,16),2)))
+                                            
+                                            self.genderDict[name][mf][country] = norm_freq
+               
+            self.count += 1
+        
+        for [name, frequencies] in self.shortNames:
+            shortName, longName = name.split()
+            if shortName in self.genderDict and not longName in self.genderDict:
+                for nameList in self.genderDict[shortName]:
+                    if longName in self.genderDict:
+                        self.genderDict[longName].append(nameList)
+                    else:
+                        self.genderDict[longName] = [nameList]
+                    
+            elif longName in self.genderDict and not shortName in self.genderDict:
+                for nameList in self.genderDict[longName]:
+                    if shortName in self.genderDict:
+                        self.genderDict[shortName].append(nameList)
+                    else:
+                        self.genderDict[shortName] = [nameList]
```

Comment:
Can you double check this? I have a feeling that the correct syntax should be [nameList,], else you end up with a list of all elements, rather than one element which happens to be a list. Check that the if and else statements yield the same type.